# SIT-307
## 1. Read the article and reproduce the results (Accuracy, Precision, Recall, F-Measure) for Chicago Crime
 ## dataset using following classification methods:
• Logistic regression
• SVM
• Naïve Bayes
• MLP
• KNN
• Random forest
• XGBoost
## These results can be found in Table 1 (Performance parameters for Chicago) of the manuscript and should be used for comparison purposes, if required. Write a report summarising the dataset, used ML methods, experiment protocol and results including variations, if any. During reproducing the results:
### v) you should use the same set of features used by the authors.
### vi) you should use the same classifier with exact parameter values.
### vii) you should use the same training/test splitting approach as used by the authors.
### viii) you should use the same pre/post processing, if any, used by the authors.

### (v) If you find any issue in reproducing results due to incomplete description of model in the provided article, then make your own assumption and explain the reason. If your justification is correct, then your solution will be considered correct and assessed accordingly.
### (vi) Similarly, variation in results due to randomness of data splitting will also be considered during evaluation based on your explanation.
### (vii) Obtained marks will be proportional to the number of ML methods that you will report in your submission with correctly reproduced results.
### (viii) Make sure your submitted Python code segment generates the reported results, otherwise you will receive zero marks for this task.

In [12]:
import pandas as pd
## Downloading the dataset on the site: https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-Presen a filter to ensure that we would only get dates lower than december 2019 was used.

dataset = pd.read_csv("Crimes_-_2001_to_Nov2021.csv")


In [13]:
dataset.info()
dataset['Date'] = pd.to_datetime(dataset['Date'])
filtered_df = dataset[dataset['Date'] <= '2019-11-02']
count = len(filtered_df)

print(count)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7032506 entries, 0 to 7032505
Data columns (total 22 columns):
 #   Column                Dtype  
---  ------                -----  
 0   ID                    int64  
 1   Case Number           object 
 2   Date                  object 
 3   Block                 object 
 4   IUCR                  object 
 5   Primary Type          object 
 6   Description           object 
 7   Location Description  object 
 8   Arrest                bool   
 9   Domestic              bool   
 10  Beat                  int64  
 11  District              float64
 12  Ward                  float64
 13  Community Area        float64
 14  FBI Code              object 
 15  X Coordinate          float64
 16  Y Coordinate          float64
 17  Year                  int64  
 18  Updated On            object 
 19  Latitude              float64
 20  Longitude             float64
 21  Location              object 
dtypes: bool(2), float64(7), int64(3), object(1

the paper is quite ambiguous and tha part where they say that the dataset contains data from chicago from 2001 to Nov 2019, they do NOT specify if it includes november or not. What is an oversight for a good understanding of the methodology and data proccess utilized. on the paper the authors mentioned the have 7019734 instances in the chicago dataset. We have 7032505 if we consider November and 7012525 if we don't what makes it difficult to properly reproduce the results. I will consider WITHOUT November because is a little bit closer to the number that the authors got and did not really mention how.


In [5]:
[
    'ID',
    'Date', 
    'Primary Type',
    'Description',
    'Location Description',
    'Year',
    'District'
]

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7011940 entries, 0 to 7011939
Data columns (total 22 columns):
 #   Column                Dtype         
---  ------                -----         
 0   ID                    int64         
 1   Case Number           object        
 2   Date                  datetime64[ns]
 3   Block                 object        
 4   IUCR                  object        
 5   Primary Type          object        
 6   Description           object        
 7   Location Description  object        
 8   Arrest                bool          
 9   Domestic              bool          
 10  Beat                  int64         
 11  District              float64       
 12  Ward                  float64       
 13  Community Area        float64       
 14  FBI Code              object        
 15  X Coordinate          float64       
 16  Y Coordinate          float64       
 17  Year                  int64         
 18  Updated On            object        
 19  

In [10]:
dataset.head()

ID Case Number       Date                Block  IUCR   Primary Type  \
0  1310933     G001704 2001-01-01    0000X W WACKER DR  0810          THEFT   
1  1432149     G156662 2001-01-01  002XX E DELAWARE PL  2825  OTHER OFFENSE   
2  1311351     G002096 2001-01-01        003XX W 40 PL  0620       BURGLARY   
3  1310541     G000008 2001-01-01     078XX S ESSEX AV  0460        BATTERY   
4  1318802     G011348 2001-01-01     124XX S STATE ST  0820          THEFT   

               Description            Location Description  Arrest  Domestic  \
0                OVER $500  PARKING LOT/GARAGE(NON.RESID.)   False     False   
1  HARASSMENT BY TELEPHONE                       RESIDENCE   False     False   
2           UNLAWFUL ENTRY  FACTORY/MANUFACTURING BUILDING   False     False   
3                   SIMPLE                       RESIDENCE   False     False   
4           $500 AND UNDER                       RESIDENCE   False      True   

   ...  Ward  Community Area  FBI Code  X Coordinate Y Coordinate  Year  \
0  ...   NaN             NaN        06     1176294.0    1902094.0  2001   
1  ...   NaN             NaN        26     1178492.0    1906652.0  2001   
2  ...   NaN             NaN        05           NaN          NaN  2001   
3  ...   NaN             NaN       08B     1194175.0    1853638.0  2001   
4  ...   NaN             NaN        06     1178560.0    1822349.0  2001   

               Updated On   Latitude  Longitude                       Location  
0  08/17/2015 03:03:40 PM  41.886704 -87.628054  (41.886703808, -87.628054456)  
1  08/17/2015 03:03:40 PM  41.899161 -87.619844  (41.899161316, -87.619843746)  
2  08/17/2015 03:03:40 PM        NaN        NaN                            NaN  
3  08/17/2015 03:03:40 PM  41.753316 -87.563985  (41.753316255, -87.563985474)  
4  08/17/2015 03:03:40 PM  41.667824 -87.622155   (41.66782394, -87.622155017)  

[5 rows x 22 columns]